### PR TITLE
UseCORS/before/UseResponseCaching

### LINKED ISSUE #18904: UseCORS must come before UseResponseCaching
From https://github.com/dotnet/aspnetcore/issues/23218

Update **Built-in middleware** and **Order** sections

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9898482c-fa54-7ce2-1716-d14fbe03990d
* Version Independent ID: 3926c686-0ccc-f9fe-ca52-0134612a1623
* Content: [ASP.NET Core Middleware](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-3.1#built-in-middleware)
* Content Source: [aspnetcore/fundamentals/middleware/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/middleware/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #18904
Internal review URLs:
* [Middleware](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-3.1&branch=pr-en-us-18910#middleware-order) search for `UseCors`.
* [Response Caching Middleware](https://review.docs.microsoft.com/en-us/aspnet/core/performance/caching/middleware?view=aspnetcore-3.1&branch=pr-en-us-18910)
* [Cors](https://review.docs.microsoft.com/en-us/aspnet/core/security/cors?view=aspnetcore-3.1&branch=pr-en-us-18910#enable-cors)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@Tratcher  I'd ask @javiercn  to review but he never responds  😈 :smiling_imp:

File: aspnetcore/fundamentals/middleware/index.md [hunk 1]
```diff
@@ -80,7 +80,9 @@ The following `Startup.Configure` method adds security-related middleware compon
 In the preceding code:
 
 * Middleware that is not added when creating a new web app with [individual users accounts](xref:security/authentication/identity) is commented out.
-* Not every middleware needs to go in this exact order, but many do. For example, `UseCors`, `UseAuthentication`, and `UseAuthorization` must go in the order shown.
+* Not every middleware needs to go in this exact order, but many do. For example:
+  * `UseCors`, `UseAuthentication`, and `UseAuthorization` must go in the order shown.
+  * `UseCors` must go before `UseResponseCaching`.
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware/index.md:85]: Link to https://github.com/dotnet/aspnetcore/issues/23218 for context. This isn't an intentional requirement, it's a bug we should fix.

File: aspnetcore/fundamentals/middleware/index.md [hunk 2]
```diff
@@ -237,7 +239,7 @@ ASP.NET Core ships with the following middleware components. The *Order* column
 | [Authentication](xref:security/authentication/identity) | Provides authentication support. | Before `HttpContext.User` is needed. Terminal for OAuth callbacks. |
 | [Authorization](xref:Microsoft.AspNetCore.Builder.AuthorizationAppBuilderExtensions.UseAuthorization*) | Provides authorization support. | Immediately after the Authentication Middleware. |
 | [Cookie Policy](xref:security/gdpr) | Tracks consent from users for storing personal information and enforces minimum standards for cookie fields, such as `secure` and `SameSite`. | Before middleware that issues cookies. Examples: Authentication, Session, MVC (TempData). |
-| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. | Before components that use CORS. |
+| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. | Before components that use CORS. `UseCORS` must come before `UseResponseCaching`.|
```
Comment (Tratcher) [inline aspnetcore/fundamentals/middleware/index.md:242]: Same link

**Comment by Tratcher** [comment]
> @Tratcher I'd ask @javiercn to review but he never responds 😈 😈

That's between you, @javiercn, and @mkArtakMSFT.

**Pull request merged**